accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/helper/QueryHelper.java

```diff
@@ -137,20 +143,16 @@ public static Set<String> getRepPolicyNodePaths(final Session session,
         return paths;
     }
 
-    static boolean hasQueryIndexForACLs(final Session session) throws RepositoryException, IOException {
-        Query query = session.getWorkspace().getQueryManager().createQuery("EXPLAIN MEASURE SELECT * FROM [rep:ACL] AS s WHERE ISDESCENDANTNODE([/])", Query.JCR_SQL2);
+    public static boolean hasQueryIndexForACLs(final Session session) throws RepositoryException, IOException {
+        Query query = session.getWorkspace().getQueryManager().createQuery(EXPLAIN_QUERY_FOR_ACL_INDEX, Query.JCR_SQL2);
         QueryResult queryResult = query.execute();
         Row row = queryResult.getRows().nextRow();
         // inspired by https://github.com/apache/jackrabbit-oak/blob/cc8adb42d89bc4625138a62ab074e7794a4d39ab/oak-jcr/src/test/java/org/apache/jackrabbit/oak/jcr/query/QueryTest.java#L1092
         String plan = row.getValue("plan").getString();
-        String costJson = plan.substring(plan.lastIndexOf('{'));
-        
-        // use jackson for JSON parsing
-        ObjectMapper mapper = new ObjectMapper();
+        LOG.debug("Execution plan and cost for {}: {}", EXPLAIN_QUERY_FOR_ACL_INDEX, plan);
+        String costJsonStr = StringUtils.substringAfter(plan, "cost:");
```

Comment:
just checking for `{` is also very unsafe - conceptually better to check for the string after `cost:`.  (`plan.substring(plan.lastIndexOf('{'))` returned only parts of the cost json)